drivers/sensor/lm77/lm77.c

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright (c) 2021 Vestas Wind Systems A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT lm77
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+#include <sys/byteorder.h>
+
+LOG_MODULE_REGISTER(lm77, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * Only compile in trigger support if enabled in Kconfig and at least one
+ * enabled lm77 devicetree node has the int-gpios property.
+ */
+#define LM77_TRIGGER_SUPPORT_INST(inst) DT_INST_NODE_HAS_PROP(inst, int_gpios) ||
+#define LM77_TRIGGER_SUPPORT \
+	(CONFIG_LM77_TRIGGER && (DT_INST_FOREACH_STATUS_OKAY(LM77_TRIGGER_SUPPORT_INST) 0))
```

Comment:
nice trick, wasn't aware of it

Reply:
I think VARGS macro can also be used to introduce the `||` operation:
```suggestion
	(CONFIG_LM77_TRIGGER && (DT_INST_FOREACH_STATUS_OKAY_VARGS(LM77_TRIGGER_SUPPORT_INST, ||) 0))
```

https://github.com/zephyrproject-rtos/zephyr/pull/37483 adds some example docs for the non-inst version